simulator/runner/bugbase.rs

```diff
@@ -0,0 +1,241 @@
+use std::{
+    collections::HashMap,
+    io::{self, Write},
+    path::PathBuf,
+    process::Command,
+};
+
+use crate::{InteractionPlan, Paths};
+
+/// A bug is a run that has been identified as buggy.
+#[derive(Clone)]
+pub(crate) enum Bug {
+    Unloaded { seed: u64 },
+    Loaded { seed: u64, plan: InteractionPlan },
+}
+
+impl Bug {
+    /// Check if the bug is loaded.
+    pub(crate) fn is_loaded(&self) -> bool {
+        match self {
+            Bug::Unloaded { .. } => false,
+            Bug::Loaded { .. } => true,
+        }
+    }
+
+    /// Get the seed of the bug.
+    pub(crate) fn seed(&self) -> u64 {
+        match self {
+            Bug::Unloaded { seed } => *seed,
+            Bug::Loaded { seed, .. } => *seed,
+        }
+    }
+}
+
+/// Bug Base is a local database of buggy runs.
+pub(crate) struct BugBase {
+    /// Path to the bug base directory.
+    path: PathBuf,
+    /// The list of buggy runs, uniquely identified by their seed
+    bugs: HashMap<u64, Bug>,
+}
+
+impl BugBase {
+    /// Create a new bug base.
+    fn new(path: PathBuf) -> Result<Self, String> {
+        let mut bugs = HashMap::new();
+        // list all the bugs in the path as directories
+        if let Ok(entries) = std::fs::read_dir(&path) {
+            for entry in entries.flatten() {
+                if entry.file_type().is_ok_and(|ft| ft.is_dir()) {
+                    let seed = entry
+                        .file_name()
+                        .to_string_lossy()
+                        .to_string()
+                        .parse::<u64>()
+                        .or(Err(format!(
+                            "failed to parse seed from directory name {}",
+                            entry.file_name().to_string_lossy()
```

Comment:
In  the future we could self host the database on a limbo database ;)

Reply:
I actually thought about doing this too :D It would be soo good.